src/Commands/Images/GenerateAbstract.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * 2019-present Friends of Presta community
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * that is bundled with this package in the file LICENSE
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/MIT
+ *
+ * @author Friends of Presta community
+ * @copyright 2019-present Friends of Presta community
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace FOP\Console\Commands\Images;
+
+use Configuration;
+use FOP\Console\Command;
+use Image;
+use ImageManager;
+use ImageType;
+use Language;
+use Module;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class GenerateAbstract extends Command
+{
+    /** @var array */
+    protected $errors = [];
+
+    /** @var OutputInterface */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+
+        return parent::initialize($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:images:generate:' . static::IMAGE_TYPE)
+            ->setDescription('Regenerate ' . static::IMAGE_TYPE . ' thumbnails')
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'images formats comma separated',
+                'all'
+            )
+            ->addOption(
+                'delete',
+                'd',
+                InputOption::VALUE_OPTIONAL,
+                'Delete current thumbnails',
+                true
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $formats = explode(',', $input->getArgument('format'));
+        $delete = (bool) $input->getOption('delete');
```

Comment:
In fact, that option require no value. It could be changed to `InputOption::VALUE_NONE`

```php
->addOption(
                'delete',
                'd',
                InputOption::VALUE_NONE,
                'Delete current thumbnails'
            );
```